master/buildbot/test/unit/test_scripts_cleanupdb.py

```diff
@@ -43,6 +43,19 @@ def mkconfig(**kwargs):
     return config
 
 
+def patch_environ(case, key, value):
+    """
+    Add an environment variable for the duration of a test.
+    """
+    old_environ = os.environ.copy()
+
+    def cleanup():
+        os.environ.clear()
+        os.environ.update(old_environ)
```

Comment:
`os.environ` is weird, and not a real dictionary. Modifying it actually modifies the running processes environment (which gets inherited by subprocesses, for example). Doing `os.environ = old_environ` would replace it with a regular dictionary that wouldn't have that behavior.